Code/Framework/AzCore/AzCore/Math/VectorN.cpp

```diff
@@ -0,0 +1,50 @@
+/*
+ * Copyright (c) Contributors to the Open 3D Engine Project.
+ * For complete copyright and license terms please see the LICENSE at the root of this distribution.
+ *
+ * SPDX-License-Identifier: Apache-2.0 OR MIT
+ *
+ */
+
+#include <AzCore/Math/VectorN.h>
+#include <AzCore/Math/MathUtils.h>
+#include <AzCore/Math/MathScriptHelpers.h>
+
+namespace AZ
+{
+    void VectorN::Reflect([[maybe_unused]] ReflectContext* context)
+    {
+        if (auto serializeContext = azrtti_cast<SerializeContext*>(context))
+        {
+            serializeContext->Class<VectorN>()
+                ->Version(1)
+                ->Field("NumElements", &VectorN::m_numElements)
+                ->Field("Values", &VectorN::m_values)
+                ;
+
+            if (EditContext* editContext = serializeContext->GetEditContext())
+            {
+                editContext->Class<VectorN>("N-Dimensional Vector", "")
+                    ->ClassElement(Edit::ClassElements::EditorData, "")
+                    ->DataElement(
+                        Edit::UIHandlers::Default, &VectorN::m_numElements, "Total Elements", "The total number of elements in the vector")
```

Comment:
This should probably have a ChangeNotify handler that resizes m_values when it changes.